powerline/selectors/tmux.py

```diff
@@ -0,0 +1,32 @@
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+from powerline.bindings.tmux import get_tmux_output
+import subprocess
+import re
+
+_cached_clients = {}
+# This Regex parses pstree -s <Process_ID> information
+# It will match, if any process name is sshd
+SSHD_REGEX = re.compile(r"(?:^|\W)sshd(?:^|\W)")
+
+
+def tmux_via_ssh(pl, segment_info, mode, caching=True, default=False):
+	'''Returns True if tmux client is connected via SSH
+	'''
+	pid = get_tmux_output(pl, "display", "-p", "#{client_pid}")
+	if not pid.isnumeric():
+		return default
+	if caching:
+		if pid in _cached_clients:
```

Comment:
This logical to write using EAFP principle: just return `_cached_clients[pid]` and if that throws `KeyError` proceed with using `psutil`.